src/main/java/org/spongepowered/common/mixin/core/network/MixinNetHandlerPlayServer.java

```diff
@@ -524,6 +525,7 @@ public void onDisconnectHandler(PlayerList this$0, ITextComponent component) {
         if (!event.isMessageCancelled()) {
             event.getChannel().ifPresent(channel -> channel.send(player, event.getMessage()));
         }
+        ((MixinEntityPlayerMP) (Object) this.player).borderListener.onPlayerDisconnect();
```

Comment:
Same here, this cast shouldn't work. You'll need to expose it using an additional interface (check `org.spongepowered.common.interfaces` for some examples)